x-pack/plugins/apm/public/components/shared/Links/apm/AnomalyDetectionSetupLink.tsx

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React from 'react';
+import { EuiButtonEmpty, EuiToolTip, EuiIcon } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { APMLink } from './APMLink';
+import { ENVIRONMENT_NOT_DEFINED } from '../../../../../common/environment_filter_values';
+import { useUrlParams } from '../../../../hooks/useUrlParams';
+import { useFetcher, FETCH_STATUS } from '../../../../hooks/useFetcher';
+
+export function AnomalyDetectionSetupLink() {
+  const { uiFilters, urlParams } = useUrlParams();
+  // check both uiFilters and urlParams for selected environment
+  const environment = uiFilters.environment || urlParams.environment;
+
+  const { data = [], status } = useFetcher(
+    (callApmApi) =>
+      callApmApi({ pathname: `/api/apm/settings/anomaly-detection` }),
+    [],
+    { preservePreviousData: false }
+  );
+  const isFetchSuccess = status === FETCH_STATUS.SUCCESS;
+
+  const hasJobs = data.length > 0;
+  const hasJobForEnv = environment
+    ? data.some(({ environment: env }) => environment === env)
+    : true;
+
+  const showAlert = isFetchSuccess && (!hasJobs || !hasJobForEnv);
```

Comment:
Maybe I'm missing something but afaict this can be simplified a lot:

```suggestion
const showAlert = data.some(job => environment === job.environment)
```

Reply:
We would still need to check `isFetchSuccess` so it doesn't flash the alert prematurely:
```typescript
const showAlert = isFetchSuccess && !data.jobs.some((job) => environment === job.environment);
```

I do like how concise it is. I intentionally wrote it this way because I wanted the 2 conditions for showing an alert to be very clear in the code (1. there are no jobs OR 2. there are no jobs matching the current environment). It's not obvious with only the `some` call that it handles the case where no jobs exist. MDN docs even warns about this case specifically: "Caution: Calling this method on an empty array returns false for any condition!"

But I agree that less code written means less code to maintain. And we can always capture the various conditions in unit tests. Thanks for the suggestion.

Reply:
I agree. We shouldn't aim to make things one liners if it hurts readability. In this case I think the shorter version _is_ the more readable though. 

And yes, I got it a little wrong. You adaptation is better:
```
const showAlert = isFetchSuccess && !data.jobs.some((job) => environment === job.environment);
```